src/vs/workbench/parts/extensions/electron-browser/extensionsActions.ts

```diff
@@ -872,6 +876,39 @@ export class ManageExtensionAction extends Action {
 	}
 }
 
+export class ExtensionInfoAction extends Action implements IExtensionAction {
+
+	static readonly ID = 'extensions.extensionInfo';
+	static LABEL = localize('extensionInfoAction', "Copy Extension info to clipboard");
+
+	private _extension: IExtension;
+	get extension(): IExtension { return this._extension; }
+	set extension(extension: IExtension) { this._extension = extension; }
+
+	constructor(label: string,
+		@INotificationService private notificationService: INotificationService
+	) {
+		super(ExtensionInfoAction.ID, label);
+	}
+
+	run(): TPromise<any> {
+		const { description, version, publisherDisplayName, id } = this.extension;
+
+		const localizedExtension = localize('extensionInfoId', 'Extension') + ': ' + id;
+		const localizedDescription = localize('extensionInfoDescription', 'Description') + ': ' + description;
+		const localizedVersion = localize('extensionInfoVersion', 'Version') + ': ' + version;
+		const localizedPublisher = localize('extensionInfoPublisher', 'Publisher') + ': ' + publisherDisplayName;
+		const localizedVSMarketplaceLink = localize('extensionInfoVSMarketplaceLink', 'VS Marketplace Link') + ': https://marketplace.visualstudio.com/items?itemName=' + id;
+
+		const clipboardStr = `${localizedExtension}\n${localizedDescription}\n${localizedVersion}\n${localizedPublisher}\n${localizedVSMarketplaceLink}`;
+
+		clipboard.writeText(clipboardStr);
+		this.notificationService.info(localize('extensionInfoActionNotification', 'Extension info copied to clipboard'));
```

Comment:
To stay consistent with other copy actions in the product that do not show any notifications on copy (For eg: `Copy Path`), let's skip the notification here.

Reply:
Done